go/vt/discovery/healthcheck.go

```diff
@@ -63,25 +63,33 @@ type HealthCheck interface {
 }
 
 // NewHealthCheck creates a new HealthCheck object.
-func NewHealthCheck(connTimeout time.Duration, retryDelay time.Duration) HealthCheck {
+func NewHealthCheck(connTimeout time.Duration, retryDelay time.Duration, healthCheckTimeout time.Duration) HealthCheck {
 	hc := &HealthCheckImpl{
-		addrToConns: make(map[string]*healthCheckConn),
-		targetToEPs: make(map[string]map[string]map[topodatapb.TabletType][]*topodatapb.EndPoint),
-		connTimeout: connTimeout,
-		retryDelay:  retryDelay,
+		addrToConns:        make(map[string]*healthCheckConn),
+		targetToEPs:        make(map[string]map[string]map[topodatapb.TabletType][]*topodatapb.EndPoint),
+		connTimeout:        connTimeout,
+		retryDelay:         retryDelay,
+		healthCheckTimeout: healthCheckTimeout,
 	}
 	if hcConnCounters == nil {
 		hcConnCounters = stats.NewMultiCountersFunc("HealthcheckConnections", []string{"keyspace", "shardname", "tablettype"}, hc.servingConnStats)
 	}
+	go func() {
+		t := time.NewTicker(healthCheckTimeout / 3)
+		for range t.C {
+			hc.checkHealthCheckTimeout()
+		}
```

Comment:
We need to add a method to terminate that timer, and free the resources for this health check? It will need to be called by the BinlogPlayerController when it is removed... in binlog.go, we call blm.players[source].Stop() around line 483, we should also close the healthcheck there. I can do that cleanup later though if you want, but I need the close method here first.